### PR TITLE
update mongoose-delete, using mongoose own dts

### PR DESCRIPTION
**Pull request by bangbang93**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/Automattic/mongoose/blob/master/CHANGELOG.md#5110--2020-11-30>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/mongoose-delete/mongoose-delete-tests.ts [hunk 1]
```diff
@@ -1,49 +1,49 @@
...
+PetSchema.plugin(MongooseDelete, {
   overrideMethods: 'all',
   deletedByType: String,
 });
-PetSchema.plugin(mongoose_delete, { overrideMethods: 'all', deletedBy: true, indexFields: ['deleted'] });
-PetSchema.plugin(mongoose_delete, { overrideMethods: 'all', deletedBy: true, indexFields: 'all' });
-PetSchema.plugin(mongoose_delete, { overrideMethods: 'all', deletedBy: true, indexFields: 'invalid' }); // $ExpectError
+PetSchema.plugin(MongooseDelete, { overrideMethods: 'all', deletedBy: true, indexFields: ['deleted'] });
+PetSchema.plugin(MongooseDelete, { overrideMethods: 'all', deletedBy: true, indexFields: 'all' });
+PetSchema.plugin(MongooseDelete, { overrideMethods: 'all', deletedBy: true, indexFields: 'invalid' });
```
Comment (weswigham) [inline types/mongoose-delete/mongoose-delete-tests.ts:34]: Why was the `// $ExpectError` removed? Shouldn't this call still issue an error?

File: types/mongoose-delete/mongoose-delete-tests.ts [hunk 2]
```diff
@@ -1,49 +1,49 @@
...
   overrideMethods: ['count', 'find', 'findOne', 'findOneAndUpdate', 'update'],
 });
 // or
-PetSchema.plugin(mongoose_delete, {
+PetSchema.plugin(MongooseDelete, {
   overrideMethods: ['count', 'countDocuments', 'find'],
 });
 // or (unrecognized method names will be ignored)
-PetSchema.plugin(mongoose_delete, { overrideMethods: ['count', 'find', 'errorXyz'] }); // $ExpectError
+PetSchema.plugin(MongooseDelete, { overrideMethods: ['count', 'find', 'errorXyz'] });
```
Comment (weswigham) [inline types/mongoose-delete/mongoose-delete-tests.ts:24]: And this one?

File: types/mongoose-delete/mongoose-delete-tests.ts (hunk 1 above)
Comment (bangbang93) [reply types/mongoose-delete/mongoose-delete-tests.ts:34]: in `@types/mongoose@5.10`, `plugin` function has an overload `plugin<T>(plugin: (schema: Schema, options: T) => void, opts: T): this;)`. So the 'opts' params can be limit to `boolean | 'all' | MongooseDelete.overridableMethods[]`. But in `mongoose`, It's `plugin(fn: (schema: Schema<DocType>, opts?: any) => void, opts?: any): this;`, is an `any`. So should not issue an error.

File: types/mongoose-delete/mongoose-delete-tests.ts (hunk 2 above)
Comment (bangbang93) [reply types/mongoose-delete/mongoose-delete-tests.ts:24]: same as above

File: types/mongoose-delete/mongoose-delete-tests.ts (hunk 1 above)
Comment (weswigham) [reply types/mongoose-delete/mongoose-delete-tests.ts:34]: Ick - that's a pretty bad UX they've regressed to, but OK.

**Review by weswigham — APPROVED** [review]

**Comment by gforge** [comment]
Thanks for trying to fix the typings. I'm having quite a few issues with the types and it would be nice if there were some more tests that show exactly how to use the types, e.g. I can't even get this simple thing to pass:

```ts
const retrievedFluffy = Pet.findOne({ name: 'fluffy' });
if (retrievedFluffy) {
    const deleted: boolean | undefined = retrievedFluffy.deleted;
    const name: string = retrievedFluffy.name;
}
```

I get the error:

```text
ERROR: 101:55  expect                   TypeScript@4.5 compile error: 
Property 'deleted' does not exist on type 'Query<(Document<any, any, PetDocument> & PetDocument & { _id: any; }) | null, Document<any, any, PetDocument> & PetDocument & { ...; }, {}, PetDocument>'.
```

**Comment by bangbang93** [comment]
> Thanks for trying to fix the typings. I'm having quite a few issues with the types and it would be nice if there were some more tests that show exactly how to use the types, e.g. I can't even get this simple thing to pass:
> 
> ```ts
> const retrievedFluffy = Pet.findOne({ name: 'fluffy' });
> if (retrievedFluffy) {
>     const deleted: boolean | undefined = retrievedFluffy.deleted;
>     const name: string = retrievedFluffy.name;
> }
> ```
> 
> I get the error:
> 
> ```
> ERROR: 101:55  expect                   TypeScript@4.5 compile error: 
> Property 'deleted' does not exist on type 'Query<(Document<any, any, PetDocument> & PetDocument & { _id: any; }) | null, Document<any, any, PetDocument> & PetDocument & { ...; }, {}, PetDocument>'.
> ```

model needs to extends `MongooseDelete.SoftDeleteModel`

**Comment by bangbang93** [comment]
Ready to merge

**Pull request merged**